libs/framework/include/celix_bundle_activator.h

```diff
@@ -139,75 +139,6 @@ celix_status_t celix_bundleActivator_destroy(void *userData, celix_bundle_contex
 
 #ifdef __cplusplus
 }
-
-
-/**
- * This macro generates the required bundle activator functions for C++.
- * This can be used to more type safe bundle activator entries.
- *
- * The macro will create the following bundle activator functions:
- * - bundleActivator_create which allocates a pointer to the provided type.
- * - bundleActivator_start/stop which will call the respectively provided typed start/stop functions.
- * - bundleActivator_destroy will free the allocated for the provided type.
- *
- * @param type The activator type (e.g. 'ShellActivator'). A type which should have a constructor with a single arugment of std::shared_ptr<DependencyManager>.
- */
-#define CELIX_GEN_CXX_BUNDLE_ACTIVATOR(actType)                                                                        \
```

Comment:
Is there any transition macro possible to define to ease refactoring cost for users that have been using this?

Reply:
Both style should be possible, thanks to std::enable_if:
https://github.com/apache/celix/blob/feature/cxx_headers/libs/framework/include/celix/BundleActivator.h#L52

Reply:
Oh, I must've missed the moving of the define. I thought it was deleted entirely.